DesignDocs.md

```diff
@@ -0,0 +1,65 @@
+
+## What Is a Design Document?
+It's a document clearly describing a problem, a solution to that problem, and why we think the proposal is the right solution.
+It has enough information to begin writing stories for implementing the solution in the document.
+ 
+## Audience
+The audience for DSNP Design Documents is mainly DSNP developers, both internal and at large.
+We also expect other developers and technically savvy readers will read these documents. 
+Linking to technical explainers is encouraged, such as Wikipedia, the DSNP Spec, or other DSNP Design Documents.
+This will help readers to learn about concepts in the proposal if need be.
+
+## Sections
+There are six required sections and one optional section.
+
+* **Context and Scope**: A short description of the landscape in which the new system is being built, what is actually being built. It may also say what is _not_ being built, and any assumptions.
+* **Problem Statement**: The "why." A short summary of the issue(s) that this design solves.
+    This doesn't have to be a technical problem. 
+    "Developer unhappiness" or "improved user experience" are also problems.
+* **Goals and Non-goals**:  what this solution is trying to do, and is also _not_ trying to do, in concrete terms.  Measurable goals are best.
+* **Proposal**: A high level overview, followed by a detailed description. 
+    This is where specific technology, such as language, frameworks, encryption algorithms, type of authentication, and APIs can be described.  
+    It can include diagrams such as [a system context diagram](https://en.wikipedia.org/wiki/System_context_diagram), or a [sequence diagram](https://www.geeksforgeeks.org/unified-modeling-language-uml-sequence-diagrams/).
+* **Benefits and Risks**: the reasons why this solution was chosen, and the risks this solution poses.
+    For example, the solution may be very simple, but there could performance bottlenecks above a certain threshold.
+    Another: the solution is well known and widely used, but it's not a perfect fit and requires complicated changes in one area.   
+* **Alternatives and Rationale**: discuss alternatives that were considered, and why they were rejected. 
+    Note when there are absolute requirements that the solution does not and can't meet.
+    One example might be, it's a proprietary solution but we need something open source.
+* **Sources**: sources of information that led to this design.
+* **Glossary** (optional): if you think inline links to concepts are too distracting, include a glossary section. 
+    This can be links, text or both.
+
+## Process
+The process for adding a new DSNP Design Document is much like any other Pull Request in a code base.
+1. Create a branch in this repo.
+2. Commit your design doc in your branch, based on the guidelines here.
+3. Post the design document as a Draft PR for discussion.
+4. When the PR is accepted, it can be merged.
```

Comment:
I think that's a good point in terms of the design doc mirroring the implementation as close as possible. I'm just thinking about situations where a doc PR is approved but some changes were needed on the actual feature.

Reply:
We'd post the Design Doc, it would be discussed, reviewed and merged, and then work would begin on it, starting with issues (stories) in the respective GitHub Project.  This would not be posted after the fact.

It doesn't preclude updates to the Design Docs, either.  I think it would be a case by case basis, but if for example it turns out that you don't need a certain API call or the call needs an extra parameter, it's updated in the Design Doc just the way a regular PR would be done in a code repo.

If say, the entire Design is scrapped because the feature is killed or something like that, we could maybe retire it by marking it retired, or something like that.  I think it's a little too early to lay out a process for that kind of thing.  We want to keep the first iteration simple and easy to pick up for now.